widgets/common/src/ExecuteDeploymentModal.jsx

```diff
@@ -51,13 +94,18 @@ export default function ExecuteDeploymentModal({
         const { InputsUtils, DeploymentActions } = Stage.Common;
         const validationErrors = {};
 
-        if (!deployment || !workflow) {
-            setErrors({ error: 'Missing workflow or deployment' });
+        const name = getWorkflowName(workflow);
+        if (!getWorkflowName(workflow)) {
```

Comment:
Looks like a half-baked refactor :smile: 

```suggestion
        if (!name) {
```